python/src/aiconfig/default_parsers/openai_vision.py

```diff
@@ -0,0 +1,427 @@
+                        attachment_data = AttachmentDataWithStringValue(
+                            kind=attachment_kind, value=image_url
+                        )
+                        input["attachments"].append(
+                            Attachment(
+                                data=attachment_data, mime_type="image/*"
+                            )
+                        )
+
+                # openai sdk currently only supports images with user messages
+                assistant_output = []
+                if assistant_response is not None:
+                    output_data = build_output_data(assistant_response)
+                    metadata = {
+                        "raw_response": assistant_response,
+                        role: "assistant",
+                    }
+                    assistant_output = [
+                        ExecuteResult(
+                            output_type="execute_result",
+                            execution_count=None,
+                            data=output_data,
+                            metadata=metadata,
+                        )
+                    ]
+                prompt = Prompt(
+                    name=new_prompt_name,
+                    input=PromptInput(
+                        data=input.get("data"),
+                        attachments=input.get("attachments"),
+                    ),
+                    metadata=PromptMetadata(
+                        model=copy.deepcopy(model_metadata),
+                        parameters=parameters,
+                        remember_chat_context=True,
+                    ),
+                    outputs=assistant_output,
+                )
+                prompts.append(prompt)
+            elif i == 0 and role == "assistant":
+                # If the first message is an assistant message,
+                # build a prompt with an empty input,
+                # and the assistant response as the output
+
+                # Pull assistant response
+                assistant_output = build_output_data(
+                    conversation_data["messages"][i]
+                )
+                prompt = Prompt(
+                    name=f"{prompt_name}_{len(prompts) + 1}",
+                    input="",
+                    metadata=PromptMetadata(
+                        model=copy.deepcopy(model_metadata),
+                        parameters=parameters,
+                        remember_chat_context=True,
+                    ),
+                    outputs=[
+                        ExecuteResult(
+                            output_type="execute_result",
+                            execution_count=None,
+                            data=assistant_output,
+                            metadata={},
+                        )
+                    ],
+                )
+                prompts.append(prompt)
+            i += 1
+
+        if prompts:
+            prompts[len(prompts) - 1].name = prompt_name
```

Comment:
Do we still need this if we do the refactoring part above for prompt name to use `i`?

Reply:
I think it's needed regardless. If my understanding is correct it's just assuming that the user is serializing a full message history and wants the latest prompt to be named exactly prompt_name instead of some suffix for uniqueness